### PR TITLE
Add functionality for snapshotter charts

### PR DESCRIPTION
**Pull request by dmcdii**
Adding network policy, pod disruption budget, volume snapshot class, topology spread constraints & other small changes

Signed-off-by: David McDaniel <david.mcdaniel@ontariosystems.com>

### CODE REVIEW

File: charts/snapshot-validation-webhook/values.yaml [hunk 1]
```diff
@@ -24,6 +24,28 @@ tls:
 imagePullSecrets: []
 podAnnotations: {}
 
+NetworkPolicy:
```
Comment (WanzenBug) [inline charts/snapshot-validation-webhook/values.yaml:27]: Convention is that this should be spelled `networkPolicy`

File: charts/snapshot-validation-webhook/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    {{- include "snapshot-validation-webhook.labels" . | nindent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "snapshot-validation-webhook.selectorLabels" . | nindent 6 }}
+  ingress:
+    - from:
+      - podSelector:
+          matchLabels:
+            {{- include "snapshot-validation-webhook.selectorLabels" . | nindent 12 }}
```
Comment (WanzenBug) [inline charts/snapshot-validation-webhook/templates/networkpolicy.yaml:16]: I don't quite get what's the point of this network policy? It allows for the pod to talk with other instances of the same deployment, but I don't see how thats useful. The only thing that should have access to the pod is the kube-apiserver (or whatever component actually calls the webhook), and maybe some monitoring.

There is also literally nothing to protect here, apart from maybe some DOS concerns by someone overloading the webhook...

File: charts/snapshot-controller/values.yaml [hunk 3]
```diff
@@ -41,3 +41,11 @@ serviceAccount:
   # Specifies whether a ServiceAccount should be created
   create: true
   name: ""
+
+volumeSnapshotClasses: []
+#  - name: ebs-csi-aws-delete
+#    annotations:
+#      snapshot.storage.kubernetes.io/is-default-class: "true"
+#    driver: ebs.csi.aws.com
+#    deletionPolicy: Delete
```
Comment (WanzenBug) [inline charts/snapshot-controller/values.yaml:50]: I don't really see this as responsibility of the snapshot-controller. Without some CSI driver you don't need any snapshot classes. Setting up snapshot classes should be the responsibility of that CSI driver deployment, not the generic snapshot-controller.

**Review by WanzenBug — CHANGES_REQUESTED** [review]
I'd like to see some changes, or at least better understanding on why certain parts are needed

File: charts/snapshot-validation-webhook/values.yaml (hunk 1 above)
Comment (dmcdii) [reply charts/snapshot-validation-webhook/values.yaml:27]: Agreed; I will correct this.

File: charts/snapshot-validation-webhook/templates/networkpolicy.yaml (hunk 2 above)
Comment (dmcdii) [reply charts/snapshot-validation-webhook/templates/networkpolicy.yaml:16]: The purpose is for clusters that utilize network policies that block all ingress by default (zero trust) the helm tests will fail without a network policy in place that allows it. The helm test pod is making a connection to the webhook pod via curl to perform the test. Without the network policy in place the tests fail for us. In our case, we utilize automated helm installs/updates and run helm tests provided by charts as the default behavior.

File: charts/snapshot-controller/values.yaml (hunk 3 above)
Comment (dmcdii) [reply charts/snapshot-controller/values.yaml:50]: The external-shapshotter is the source of record for the CRD definition. I set the default of the chart not to create snapshot classes. I have included the commented out lines as an example. In general CSI drivers can function without a snapshot controller so they may not want to include a feature for creating resources for a external/optional dependency. I included it for convenience so a separate chart would not be needed. If you feel strongly it should not be here, I can remove it.

File: charts/snapshot-validation-webhook/templates/networkpolicy.yaml (hunk 2 above)
Comment (WanzenBug) [reply charts/snapshot-validation-webhook/templates/networkpolicy.yaml:16]: Well, I guess it can't hurt.

**Review by WanzenBug — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**